### PR TITLE
Azure: parameterize ssh user name

### PR DESCRIPTION
**Pull request by kartikjoshi21**
change default value of ssh user name to peerpod and parameterize it in case user wants to provide custom name

Fixes: #717

### CODE REVIEW

**Review by surajssd — COMMENTED** [review]
Right now this is exposed at the flag level, but this should bubble up as an env var to the kustomization file here: https://github.com/confidential-containers/cloud-api-adaptor/blob/53201a825540127507e1d2a390eb8b40e379ef67/install/overlays/azure/kustomization.yaml#L19, isn't it?

**Review by knrt10 — APPROVED** [review]

**Comment by surajssd** [comment]
I think this is still missing changing the username at the packer level.

**Review by surajssd — CHANGES_REQUESTED** [review]
I build a new image and tried deploying it but it still uses older username:

```bash
# ssh -i /tmp/key peerpod@10.2.0.5
Warning: Permanently added '10.2.0.5' (ED25519) to the list of known hosts.
peerpod@10.2.0.5: Permission denied (publickey).

# ssh -i /tmp/key ubuntu@10.2.0.5
Welcome to Ubuntu 22.04.2 LTS (GNU/Linux 5.15.0-1035-azure x86_64)

...

ubuntu@podvm-nginx-86fcc9854-8cjzx-9046ce1c:~$ ps aufx | grep nginx
1001         799  0.0  0.0  24172  5472 ?        S    17:34   0:00  \_ nginx: master process /opt/bitnami/nginx/sbin/nginx -c /opt/bitnami/nginx/conf/nginx.conf -g daemon off;
1001         823  0.0  0.0  24612  2436 ?        S    17:34   0:00      \_ nginx: worker process
1001         824  0.0  0.0  24612  2436 ?        S    17:34   0:00      \_ nginx: worker process
1001         825  0.0  0.0  24612  2436 ?        S    17:34   0:00      \_ nginx: worker process
1001         826  0.0  0.0  24612  2436 ?        S    17:34   0:00      \_ nginx: worker process
1001         827  0.0  0.0  24612  2436 ?        S    17:34   0:00      \_ nginx: worker process
1001         828  0.0  0.0  24612  2436 ?        S    17:34   0:00      \_ nginx: worker process
1001         829  0.0  0.0  24612  2436 ?        S    17:34   0:00      \_ nginx: worker process
1001         830  0.0  0.0  24612  2436 ?        S    17:34   0:00      \_ nginx: worker process
ubuntu       990  0.0  0.0   4020  2040 pts/0    S+   17:39   0:00              \_ grep --color=auto nginx
```

**Comment by kartikjoshi21** [comment]
**Test with default user peerpod**

```
root@security-context-demo:/# ssh -i id_rsa 10.2.0.8
The authenticity of host '10.2.0.8 (10.2.0.8)' can't be established.
ECDSA key fingerprint is SHA256:8CXyKnbfA5HDJbLrFcRhNDBDHu+ntpYszUMXqN8ExSU.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added '10.2.0.8' (ECDSA) to the list of known hosts.
Enter passphrase for key 'id_rsa': 
Please login as the user "peerpod" rather than the user "root".

Connection to 10.2.0.8 closed.
root@security-context-demo:/# ssh -i id_rsa peerpod@10.2.0.8
Enter passphrase for key 'id_rsa': 
Welcome to Ubuntu 22.04.2 LTS (GNU/Linux 5.15.0-1035-azure x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage


This system has been minimized by removing packages and content that are
not required on a system that users do not log into.

To restore this content, you can run the 'unminimize' command.

Expanded Security Maintenance for Applications is not enabled.

0 updates can be applied immediately.

Enable ESM Apps to receive additional future security updates.
See https://ubuntu.com/esm or run: sudo pro status


The list of available updates is more than a week old.
To check for new updates run: sudo apt update

The programs included with the Ubuntu system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Ubuntu comes with ABSOLUTELY NO WARRANTY, to the extent permitted by
applicable law.

To run a command as administrator (user "root"), use "sudo <command>".
See "man sudo_root" for details.

peerpod@podvm-nginx-deployment-86fcc9854-5x4ks-bb784602:~$ 
```

**Test with username kartikjoshi**
export SSH_USERNAME=kartikjoshi
```
root@security-context-demo:/# ssh -i id_rsa 10.2.0.9
The authenticity of host '10.2.0.9 (10.2.0.9)' can't be established.
ECDSA key fingerprint is SHA256:Xk8wBBET/LoQXv+GxuhzgBbfvXYSHnL1MZVwiJmrZZg.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added '10.2.0.9' (ECDSA) to the list of known hosts.
Enter passphrase for key 'id_rsa': 
Please login as the user "kartikjoshi" rather than the user "root".

Connection to 10.2.0.9 closed.
root@security-context-demo:/# ssh -i id_rsa kartikjoshi@10.2.0.9
Enter passphrase for key 'id_rsa': 
Welcome to Ubuntu 22.04.2 LTS (GNU/Linux 5.15.0-1035-azure x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage


This system has been minimized by removing packages and content that are
not required on a system that users do not log into.

To restore this content, you can run the 'unminimize' command.

Expanded Security Maintenance for Applications is not enabled.

0 updates can be applied immediately.

Enable ESM Apps to receive additional future security updates.
See https://ubuntu.com/esm or run: sudo pro status


The list of available updates is more than a week old.
To check for new updates run: sudo apt update

The programs included with the Ubuntu system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Ubuntu comes with ABSOLUTELY NO WARRANTY, to the extent permitted by
applicable law.

kartikjoshi@podvm-nginx-deployment-86fcc9854-ztr9w-c6516295:~$ 
```

File: azure/image/azure-ubuntu.pkr.hcl [hunk 1]
```diff
@@ -48,12 +48,23 @@ build {
...
 
+  provisioner "file" {
+    source      = "create-user.sh"
+    destination = "~/create-user.sh"
+  }
+
+  provisioner "shell" {
+    remote_folder = "~"
+    inline = [
+      "sudo bash ~/create-user.sh ${var.ssh_username}"
```
Comment (surajssd) [inline azure/image/azure-ubuntu.pkr.hcl:59]:
```suggestion
      "sudo useradd -m -s /bin/bash ${var.ssh_username}"
```

File: azure/image/create-user.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+ssh_username="peerpod"
+
+if [ -n "$1" ]; then
+    ssh_username=$1
+fi
+
+sudo useradd -m -s /bin/bash "${ssh_username}"
```
Comment (surajssd) [inline azure/image/create-user.sh:7]:
```suggestion
```

File: azure/image/azure-ubuntu.pkr.hcl (hunk 1 above)
Comment (surajssd) [reply azure/image/azure-ubuntu.pkr.hcl:59]: I think for single liner we can skip using bash script.

File: azure/image/create-user.sh (hunk 2 above)
Comment (surajssd) [reply azure/image/create-user.sh:7]: IMHO the default username setting here is redundant it is already set in the packer variables file.

**Review by surajssd — APPROVED** [review]

**Pull request merged**